cmd/discovery/main.go

```diff
@@ -38,10 +39,23 @@ func main() {
 	logs.InitLogs()
 	defer logs.FlushLogs()
 
-	cfg, err := rest.InClusterConfig()
-	if err != nil {
-		glog.Fatalf("failed to get config: %v", err)
+	var (
+		cfg *rest.Config
+		err error
+	)
+	kubeConfigFile := os.Getenv("KUBECONFIG")
+	if kubeConfigFile != "" {
+		cfg, err = clientcmd.BuildConfigFromFlags("", kubeConfigFile)
+		if err != nil {
+			glog.Fatalf("failed to create config from file: %s %v\n", kubeConfigFile, err)
+		}
+	} else {
+		cfg, err = rest.InClusterConfig()
+		if err != nil {
+			glog.Fatalf("failed to get config: %v", err)
+		}
 	}
```

Comment:
Can you extract this logic into a method `NewKubeCli(kubeCfg string) kubernetes.Interface` and then reference this method to initialize the k8s client? In addition, the client of the following CR `versioned.NewForConfig` should be initialized in the same way.

Reply:
I have get the logic into a new method which return *rest.config